pkg/detectors/alegra/alegra.go

```diff
@@ -60,7 +60,7 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			}
 
 			if verify {
-				req, err := http.NewRequestWithContext(ctx, "GET", "https://api.alegra.com/api/v1/users", nil)
+				req, err := http.NewRequestWithContext(ctx, "GET", "https://api.alegra.com/api/v1/users/self", nil)
```

Comment:
I created an account on [Alegra](https://mi.alegra.com) and generated a token. Initially, I tested the `/users` API endpoint using an API client, and it worked as expected, returning the same response as `/users/self`. However, the Alegra documentation only mentions the `/users/self` endpoint, and there’s no reference to `/users`, so I'm unsure why it’s functioning similarly.

Additionally, I noticed an issue with the detector failing to identify my user ID, which I believe is always the email used during signup. The regex used for detecting IDs has a length constraint of 25 to 30 characters. However, I was able to create an account with an email shorter than this limit, and despite using that email (as my username) along with my token to successfully call the API via the API client, the detector did not pick it up.

I suggest testing and adjusting the regex for detecting user IDs to ensure it covers valid email lengths.

As for the API behavior, I’ll leave that to @zricethezav and @abmussani. Both `/users` and `/users/self` worked for me, though only the `/users/self` endpoint is mentioned in the documentation.

Reply:
Thanks for the detailed review, @kashifkhan0771 !

I have expanded the email regex to include all case for any possible email.

Reply:
I would feel more confident approving this if you could also test the current email regex to confirm that it isn't working correctly(I tested it but a second round of testing from the PR owner will be good I believe. You can create an account and test this email regex change). A screenshot or any form of documentation from your testing would be sufficient.

Reply:
I have added the testing screenshot, please check

Reply:
New Changes are picking up the key and ID correctly now and verifying. Wait for @zricethezav and @abmussani to respond about the API change.

Reply:
Can you apply the label for Hacktoberfest @kashifkhan0771?

Reply:
Once the PR is approved. We can add it 😃 - Once again thanks for the all the fixes and contributions!